### PR TITLE
HAAR-1606 view client end point

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientService.kt [hunk 1]
```diff
@@ -48,8 +49,11 @@ class ClientService(
 
   @Transactional
   fun editClientCredentials(clientId: String, clientDetails: ClientCredentialsUpdateRequest) {
-    val existingClient = clientRepository.findClientByClientId(clientId) ?: throw ClientNotFoundException(Client::class.simpleName, clientId)
-    val existingClientConfig = clientConfigRepository.findByIdOrNull(baseClientId.toBase(clientId)) ?: throw ClientNotFoundException(ClientConfig::class.simpleName, clientId)
+    // TODO authorities update !
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientService.kt:52]: HAAR-1616 refers

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/data/model/Client.kt [hunk 2]
```diff
@@ -48,7 +49,16 @@ data class Client(
   @Column(length = 2000)
   @Convert(converter = TokenSettingsConverter::class)
   var tokenSettings: TokenSettings,
-)
+) {
+
+  fun getDatabaseUserName(): String? {
+    return tokenSettings.settings[RegisteredClientAdditionalInformation.DATABASE_USER_NAME_KEY] as String?
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/data/model/Client.kt:55]: consider static imports of contacts

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt [hunk 3]
```diff
@@ -29,6 +31,44 @@ class ClientsController(
   fun editClient(@PathVariable clientId: String, @RequestBody clientDetails: ClientCredentialsUpdateRequest) {
     clientService.editClientCredentials(clientId, clientDetails)
   }
+
+  @GetMapping("clients/client-credentials/{clientId}/view")
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt:35]: swagger docs required here?

**Review by ravmoj — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt (hunk 3 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt:35]: No, these end points are only intended for use by the new UI

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/data/model/Client.kt (hunk 2 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/data/model/Client.kt:55]: Will amend in next ticket

**Pull request merged**